pkg/dao/types/workflow.go

```diff
@@ -17,9 +18,50 @@ const (
 // RetryStrategy is the retry strategy of a workflow step.
 // See https://raw.githubusercontent.com/argoproj/argo-workflows/master/examples/retry-conditional.yaml
 type RetryStrategy struct {
-	Limit       int                  `json:"limit"`
-	RetryPolicy v1alpha1.RetryPolicy `json:"retryPolicy"`
-	Backoff     *v1alpha1.Backoff    `json:"backoff"`
+	Limit       int                     `json:"limit"`
+	RetryPolicy apiworkflow.RetryPolicy `json:"retryPolicy"`
+	Backoff     *apiworkflow.Backoff    `json:"backoff"`
+}
+
+// ParameterConfig is the config of a workflow parameter.
+// Parameters could be reconfigured in workflow execution.
+type ParameterConfig struct {
+	// Name is the name of the parameter.
+	Name        string `json:"name"`
+	Value       string `json:"value"`
+	Overwrite   bool   `json:"overwrite"`
+	Description string `json:"description,omitempty"`
+}
+
+func (c *ParameterConfig) Validate() error {
+	if c.Name == "" {
+		return errors.New("invalid input: empty name")
+	}
+
+	if c.Value == "" {
+		return errors.New("invalid input: empty value")
+	}
+
+	// Mustache syntax is not allowed.
+	// Argo workflow will try parse the value as a template parameter like
+	// {{workflow.parameters.name}}.
```

Comment:
Can we call it variables and use it in the format ${workflow.var.name}?

Reply:
updated